R/add_keep_population_flag.R

```diff
@@ -0,0 +1,163 @@
+#' @family individual_file
+
+add_keep_population_flag <- function(individual_file, year) {
+  calendar_year <- paste0("20", substr(year, 1, 2)) %>% as.integer()
+
+  if (!check_year_valid(year, "NSU")) {
+    individual_file <- individual_file %>%
+      dplyr::mutate(keep_population = 1L)
+  } else {
+    ## Obtain the population estimates for Locality AgeGroup and Gender.
+    pop_estimates <-
+      readRDS(get_datazone_pop_path("DataZone2011_pop_est_2011_2021.rds")) %>%
+      dplyr::as_tibble() %>%
+      dplyr::select(year, datazone2011, sex, age0:age90plus)
+
+    # Step 1: Obtain the population estimates for Locality, AgeGroup, and Gender
+    # Select out the estimates for the year of interest.
+    # if we don't have estimates for this year (and so have to use previous year).
+    year_available <- pop_estimates %>%
+      dplyr::pull(year) %>%
+      unique()
+    if (calendar_year %in% year_available) {
+      pop_estimates <- pop_estimates %>%
+        dplyr::filter(year == calendar_year)
+    } else {
+      previous_year <- sort(year_available, decreasing = TRUE)[1]
+      pop_estimates <- pop_estimates %>%
+        dplyr::filter(year = previous_year)
+    }
+
+    pop_estimates <- pop_estimates %>%
+      # Recode gender to make it match source.
+      dplyr::mutate(sex = dplyr::if_else(sex == "M", 1, 2)) %>%
+      dplyr::rename(
+        "age90" = "age90plus",
+        "gender" = "sex"
+      ) %>%
+      tidyr::pivot_longer(
+        names_to = "age",
+        names_prefix = "age",
+        values_to = "population_estimate",
+        cols = "age0":"age90"
+      ) %>%
+      dplyr::mutate(
+        age = as.integer(age),
+        age_group = dplyr::case_when(
+          age >= 0 & age <= 4 ~ "0-4",
+          age >= 5 & age <= 14 ~ "5-14",
+          age >= 15 & age <= 24 ~ "15-24",
+          age >= 25 & age <= 34 ~ "25-34",
+          age >= 35 & age <= 44 ~ "35-44",
+          age >= 45 & age <= 54 ~ "45-54",
+          age >= 55 & age <= 64 ~ "55-64",
+          age >= 65 & age <= 74 ~ "65-74",
+          age >= 75 & age <= 84 ~ "75-84",
+          age >= 85 ~ "85+"
+        )
+      ) %>%
+      dplyr::left_join(
+        get_locality_path() %>%
+          readRDS() %>%
+          dplyr::select("locality" = "hscp_locality", datazone2011),
+        by = "datazone2011"
+      ) %>%
+      dplyr::group_by(locality, age_group, gender) %>%
+      dplyr::summarize(population_estimate = sum(population_estimate)) %>%
+      dplyr::ungroup()
+
+    # Step 2: Work out the current population sizes in the SLF for Locality, AgeGroup, and Gender
+    # Work out the current population sizes in the SLF for Locality AgeGroup and Gender.
+    individual_file <- slfhelper::read_slf_individual(year,
+      columns = c(
+        "chi",
+        "locality",
+        "age",
+        "gender",
+        # "nsu",
+        "death_date"
+      )
+    ) %>%
+      dplyr::mutate(nsu = 0L) # delete this before merge
```

Comment:
Here you are reading the individual file from hscdiip when this should be the current individual file at the point of processing. This used to be called `temp4` but you have set the data parameter as `individual_file` which you can use instead as we want the estimates from the new file we are creating
```suggestion
    individual_file <- individual_file 
```